datafusion/physical-expr/src/scalar_function.rs

```diff
@@ -44,12 +44,12 @@ use arrow::record_batch::RecordBatch;
 use datafusion_common::Result;
 use datafusion_expr::{
     expr_vec_fmt, BuiltinScalarFunction, ColumnarValue, FuncMonotonicity,
-    ScalarFunctionImplementation,
+    ScalarFunctionDefinition, ScalarFunctionImplementation,
 };
 
 /// Physical expression of a scalar function
 pub struct ScalarFunctionExpr {
-    fun: ScalarFunctionImplementation,
+    fun: ScalarFunctionDefinition,
```

Comment:
I think the `fun` here can be a `ScalarFunctionDefinition` to get the `ScalarUDF` from `ScalarFunctionExpr`?
Also I searched the code base and found ScalarFunctionImplementation is only used for:
1. BuiltinScalarFunction
2. method like `create_udf` which I think will be deprecated in the future given the new `ScalarUDFImpl` way.
So I think `ScalarFunctionImplementation` will also be deleted in a day. 🤔 WDYT?

Reply:
yeah, this should be `ScalarFunctionDefinition` here. I don't think `create_udf` will be deprecated but we have `ScalarUdfLegacyWrapper` to wrap `ScalarFunctionImplementation` and create a `ScalarUDFImpl` out of it. So the new `ScalarFunctionDefinition` would include anything that is represented as a `ScalarFunctionImplementation`